ckanext/canada/view.py

```diff
@@ -579,6 +580,72 @@ def type_redirect(resource_name, dataset_id=None):
                                    owner_org=owner_org_name))
 
 
+@canada_views.route('/recombinant/delete-selected-records/<resource_id>', methods=['GET', 'POST'])
+def delete_selected_records(resource_id):
+    lc = LocalCKAN(username=g.user)
+
+    if request.method != 'POST' or not h.check_access('datastore_records_delete',
```

Comment:
You will want to move the `request.method` condition to below. Condition the `request.method == 'POST'` along with the `'cancel' in request.form` stuff. E.g. `if request.method == 'POST' and 'confirm' in request.form ...then do the actual delete`